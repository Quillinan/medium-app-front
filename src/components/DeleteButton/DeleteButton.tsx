import { deletePost } from '@services/DeletePost/DeletePost';
import React from 'react';
import Swal from 'sweetalert2';
import { useNavigate } from 'react-router-dom';

interface DeleteButtonProps {
  postId: string;
  onDeleteSuccess: () => void;
}

const DeleteButton: React.FC<DeleteButtonProps> = ({
  postId,
  onDeleteSuccess,
}) => {
  const navigate = useNavigate();

  const handleDelete = async () => {
    const confirmation = await Swal.fire({
      title: 'Tem certeza?',
      text: 'Você realmente quer deletar este post?',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Sim, deletar',
      cancelButtonText: 'Cancelar',
    });

    if (confirmation.isConfirmed) {
      const result = await deletePost(postId);

      if (!result) {
        Swal.fire({
          title: 'Sucesso',
          text: 'Post deletado com sucesso!',
          icon: 'success',
        }).then(() => {
          onDeleteSuccess();
          navigate('/');
        });
      } else {
        console.error('Erro ao excluir o post:', result);
        Swal.fire({
          title: 'Erro',
          text: result.message || 'Não foi possível excluir o post.',
          icon: 'error',
        });
      }
    }
  };

  return (
    <button
      onClick={handleDelete}
      className='bg-red-500 text-white rounded px-4 py-1'
    >
      Apagar
    </button>
  );
};

export default DeleteButton;
