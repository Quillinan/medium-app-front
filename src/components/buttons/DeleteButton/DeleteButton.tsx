import { deletePost } from '@services/DeletePost/DeletePost';
import React from 'react';
import Swal from 'sweetalert2';

interface DeleteButtonProps {
  postId: string;
  onDeleteSuccess: () => void;
}

const DeleteButton: React.FC<DeleteButtonProps> = ({
  postId,
  onDeleteSuccess,
}) => {
  const handleDelete = async () => {
    const confirmation = await Swal.fire({
      title: 'Tem certeza?',
      text: 'Você realmente quer deletar este post?',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonText: 'Sim, deletar',
      cancelButtonText: 'Cancelar',
    });

    if (confirmation.isConfirmed) {
      const result = await deletePost(postId);

      if (!result) {
        Swal.fire({
          title: 'Sucesso',
          text: 'Post deletado com sucesso!',
          icon: 'success',
        }).then(() => {
          onDeleteSuccess();
          window.location.reload();
        });
      } else {
        console.error('Erro ao excluir o post:', result);
        Swal.fire({
          title: 'Erro',
          text: result.message || 'Não foi possível excluir o post.',
          icon: 'error',
        });
      }
    }
  };

  return (
    <button
      onClick={handleDelete}
      className='bg-red-500 text-white rounded px-4 py-1'
    >
      Apagar
    </button>
  );
};

export default DeleteButton;
