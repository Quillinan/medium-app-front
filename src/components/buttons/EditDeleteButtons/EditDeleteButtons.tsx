import DeleteButton from '@components/buttons/DeleteButton/DeleteButton';

interface EditDeleteButtonsProps {
  isEditing: boolean;
  onToggleEdit: () => void;
  onSave: () => void;
  postId: string;
}

const EditDeleteButtons: React.FC<EditDeleteButtonsProps> = ({
  isEditing,
  onToggleEdit,
  onSave,
  postId,
}) => {
  return (
    <div className='w-1/2 flex justify-end space-x-4 mt-6'>
      {isEditing ? (
        <>
          <button
            className='bg-red-500 text-white rounded px-4 py-1'
            onClick={onToggleEdit}
          >
            Cancelar
          </button>
          <button
            className='bg-blue-500 text-white rounded px-4 py-1'
            onClick={onSave}
          >
            Salvar
          </button>
        </>
      ) : (
        <>
          <button
            className='bg-yellow-500 text-white rounded px-4 py-1'
            onClick={onToggleEdit}
          >
            Editar
          </button>
          <DeleteButton postId={postId} onDeleteSuccess={() => {}} />
        </>
      )}
    </div>
  );
};

export default EditDeleteButtons;
