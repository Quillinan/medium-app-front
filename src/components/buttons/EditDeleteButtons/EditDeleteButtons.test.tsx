import { render, screen, fireEvent } from '@testing-library/react';
import EditDeleteButtons from '@components/buttons/EditDeleteButtons/EditDeleteButtons';
import DeleteButton from '@components/buttons/DeleteButton/DeleteButton';
import { vi, Mock } from 'vitest';

vi.mock('@components/buttons/DeleteButton/DeleteButton');

describe('EditDeleteButtons', () => {
  const mockOnToggleEdit = vi.fn();
  const mockOnSave = vi.fn();

  beforeEach(() => {
    mockOnToggleEdit.mockClear();
    mockOnSave.mockClear();
  });

  it('renders Edit and Delete buttons when not in editing mode', () => {
    (DeleteButton as Mock).mockImplementation(({ onDeleteSuccess }) => (
      <button onClick={onDeleteSuccess}>Mocked DeleteButton</button>
    ));

    render(
      <EditDeleteButtons
        isEditing={false}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );

    expect(screen.getByText('Editar')).toBeInTheDocument();
    expect(screen.getByText('Mocked DeleteButton')).toBeInTheDocument();
  });

  it('renders Cancelar and Salvar buttons when in editing mode', () => {
    render(
      <EditDeleteButtons
        isEditing={true}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );

    expect(screen.getByText('Cancelar')).toBeInTheDocument();
    expect(screen.getByText('Salvar')).toBeInTheDocument();
  });

  it('calls onToggleEdit when Edit button is clicked', () => {
    render(
      <EditDeleteButtons
        isEditing={false}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );

    fireEvent.click(screen.getByText('Editar'));
    expect(mockOnToggleEdit).toHaveBeenCalledTimes(1);
  });

  it('calls onToggleEdit when Cancelar button is clicked', () => {
    render(
      <EditDeleteButtons
        isEditing={true}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );

    fireEvent.click(screen.getByText('Cancelar'));
    expect(mockOnToggleEdit).toHaveBeenCalledTimes(1);
  });

  it('calls onSave when Salvar button is clicked', () => {
    render(
      <EditDeleteButtons
        isEditing={true}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );

    fireEvent.click(screen.getByText('Salvar'));
    expect(mockOnSave).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <EditDeleteButtons
        isEditing={true}
        onToggleEdit={mockOnToggleEdit}
        onSave={mockOnSave}
        postId='1'
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
