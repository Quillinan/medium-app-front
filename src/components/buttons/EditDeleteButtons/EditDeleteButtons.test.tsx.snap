// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditDeleteButtons > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-1/2 flex justify-end space-x-4 mt-6"
  >
    <button
      class="bg-red-500 text-white rounded px-4 py-1"
    >
      Cancelar
    </button>
    <button
      class="bg-blue-500 text-white rounded px-4 py-1"
    >
      Salvar
    </button>
  </div>
</DocumentFragment>
`;
